lib/matplotlib/backends/backend_agg.py

```diff
@@ -628,28 +628,25 @@ def print_tif(self, filename_or_obj, *args, **kwargs):
             """
             Other Parameters
             ----------------
-            compressed : bool
-                If present, indicates that this image
-                should be stored using the LZW compression algorithm.
+            compression : 'tiff_deflate', 'tiff_adobe_deflate', 'tiff_lzw', 
+                and other values allowed by LIBTIFF_CORE.
+                If present, indicates the compression algorithm that
+                will be passed to the backend to save the image.
             """
             buf, size = self.print_to_buffer()
             if kwargs.pop("dryrun", False):
                 return
             image = Image.frombuffer('RGBA', size, buf, 'raw', 'RGBA', 0, 1)
             dpi = (self.figure.dpi, self.figure.dpi)
-            #add TIFF compression support
-            compressed = kwargs.pop("compressed", False)
-            if compressed:
-                libtiff_original_value = TiffImagePlugin.WRITE_LIBTIFF
-                TiffImagePlugin.WRITE_LIBTIFF = True    
-                return_value = image.save(filename_or_obj, format='tiff',
-                                          dpi=dpi, compression='tiff_lzw')
-                TiffImagePlugin.WRITE_LIBTIFF = libtiff_original_value
-                return return_value
+            #add TIFF compression support by passing the parameter to backend
+            compressed = kwargs.pop("compression", None)
+            if compression is not None:
```

Comment:
don't need an if/else anymore, since compression=None is a valid arg to pass through, so you can just do:
`return image.save(filename_or_obj, format='tiff', dpi=dpi, compression=compression)`